utility/context.go

```diff
@@ -3,147 +3,124 @@ package utility
+	u.persistenceContext = nil
 	return nil
 }
 
-func (u *UtilityContext) GetLatestBlockHeight() (int64, typesUtil.Error) {
-	height, er := u.Store().GetHeight()
-	if er != nil {
-		return 0, typesUtil.ErrGetHeight(er)
-	}
-	return height, nil
-}
-
-func (u *UtilityContext) GetStoreAndHeight() (*Context, int64, typesUtil.Error) {
+// TECHDEBT: We should be using the height of the context and shouldn't need to be retrieving
+//           the height from the store either for "current height" operations.
+func (u *utilityContext) getStoreAndHeight() (modules.PersistenceRWContext, int64, error) {
 	store := u.Store()
-	height, er := store.GetHeight()
-	if er != nil {
-		return nil, 0, typesUtil.ErrGetHeight(er)
-	}
-	return store, height, nil
-}
-
-func (u *UtilityContext) Codec() codec.Codec {
-	return codec.GetCodec()
+	height, err := store.GetHeight()
+	return store, height, err
 }
 
-func (u *UtilityContext) RevertLastSavePoint() typesUtil.Error {
-	if len(u.Context.SavePointsM) == typesUtil.ZeroInt {
+// TODO: This has not been tested or investigated in detail
+func (u *utilityContext) revertLastSavePoint() typesUtil.Error {
+	if len(u.savePointsSet) == typesUtil.ZeroInt {
 		return typesUtil.ErrEmptySavePoints()
 	}
 	var key []byte
-	popIndex := len(u.Context.SavePoints) - 1
-	key, u.Context.SavePoints = u.Context.SavePoints[popIndex], u.Context.SavePoints[:popIndex]
-	delete(u.Context.SavePointsM, hex.EncodeToString(key))
-	if err := u.Context.PersistenceRWContext.RollbackToSavePoint(key); err != nil {
+	popIndex := len(u.savePointsList) - 1
+	key, u.savePointsList = u.savePointsList[popIndex], u.savePointsList[:popIndex]
+	delete(u.savePointsSet, hex.EncodeToString(key))
+	if err := u.persistenceContext.RollbackToSavePoint(key); err != nil {
 		return typesUtil.ErrRollbackSavePoint(err)
 	}
 	return nil
 }
 
-func (u *UtilityContext) NewSavePoint(transactionHash []byte) typesUtil.Error {
-	if err := u.Context.PersistenceRWContext.NewSavePoint(transactionHash); err != nil {
+//nolint:unused // TODO: This has not been tested or investigated in detail
+func (u *utilityContext) newSavePoint(transactionHash []byte) typesUtil.Error {
+	if err := u.persistenceContext.NewSavePoint(transactionHash); err != nil {
 		return typesUtil.ErrNewSavePoint(err)
 	}
 	txHash := hex.EncodeToString(transactionHash)
-	if _, exists := u.Context.SavePointsM[txHash]; exists {
+	if _, exists := u.savePointsSet[txHash]; exists {
 		return typesUtil.ErrDuplicateSavePoint()
 	}
-	u.Context.SavePoints = append(u.Context.SavePoints, transactionHash)
-	u.Context.SavePointsM[txHash] = struct{}{}
+	u.savePointsList = append(u.savePointsList, transactionHash)
+	u.savePointsSet[txHash] = struct{}{}
 	return nil
 }
 
-func (u *UtilityContext) GetBus() modules.Bus {
+func (u *utilityContext) getBus() modules.Bus {
 	return u.bus
 }
 
-func (u *UtilityContext) WithBus(bus modules.Bus) *UtilityContext {
+func (u *utilityContext) withBus(bus modules.Bus) *utilityContext {
```

Comment:
oh I must have missed these guys. `withBus`? why not `setBus` since we are here?